http://git.savannah.gnu.org/gitweb/?p=patch.git;a=commitdiff;h=91e027ab1af51717f9229d07901158e7466fcd6f

From 91e027ab1af51717f9229d07901158e7466fcd6f Mon Sep 17 00:00:00 2001
From: Andreas Gruenbacher <agruen@suse.de>
Date: Fri, 15 Jan 2010 19:20:57 +0100
Subject: [PATCH] No longer use "continue" as the only instruction in a loop body

* src/bestmatch.h, src/diffseq.h, src/inp.c, src/maketime.c,
src/merge.c, src/partime.c, src/pch.c, src/util.c: No longer use
"continue" as the only instruction in a loop body: at least one
compiler chokes on this in combination with variable-length arrays
as in src/bestmatch.h.
---
 ChangeLog       |    8 ++++++++
 src/bestmatch.h |    2 +-
 src/diffseq.h   |    4 ++--
 src/inp.c       |    4 ++--
 src/maketime.c  |    2 +-
 src/merge.c     |    4 ++--
 src/partime.c   |    2 +-
 src/pch.c       |   22 +++++++++++-----------
 src/util.c      |    6 +++---
 9 files changed, 31 insertions(+), 23 deletions(-)

diff --git a/src/bestmatch.h b/src/bestmatch.h
index 4be19d1..7c05527 100644
--- a/src/bestmatch.h
+++ b/src/bestmatch.h
@@ -131,7 +131,7 @@ bestmatch(OFFSET xoff, OFFSET xlim, OFFSET yoff, OFFSET ylim,
 		for (y = x - d;
 		     x < xlim && y < ylim && EQUAL_IDX (x, y);
 		     x++, y++)
-		  continue;
+		  /* do nothing */ ;
 		fd[d] = x;
 		if (x == xlim && y >= min
 		    && x + y - c >= fmid_plus_2_min)
diff --git a/src/diffseq.h b/src/diffseq.h
index 5ae93fa..b3d6942 100644
--- a/src/diffseq.h
+++ b/src/diffseq.h
@@ -212,7 +212,7 @@ diag (OFFSET xoff, OFFSET xlim, OFFSET yoff, OFFSET ylim, bool find_minimal,
 	  for (x = x0, y = x0 - d;
 	       x < xlim && y < ylim && EQUAL_IDX (ctxt, x, y);
 	       x++, y++)
-	    continue;
+	    /* do nothing */ ;
 	  if (x - x0 > SNAKE_LIMIT)
 	    big_snake = true;
 	  fd[d] = x;
@@ -245,7 +245,7 @@ diag (OFFSET xoff, OFFSET xlim, OFFSET yoff, OFFSET ylim, bool find_minimal,
 	  for (x = x0, y = x0 - d;
 	       xoff < x && yoff < y && EQUAL_IDX (ctxt, x - 1, y - 1);
 	       x--, y--)
-	    continue;
+	    /* do nothing */ ;
 	  if (x0 - x > SNAKE_LIMIT)
 	    big_snake = true;
 	  bd[d] = x;
diff --git a/src/inp.c b/src/inp.c
index 192cd3a..fd9517b 100644
--- a/src/inp.c
+++ b/src/inp.c
@@ -374,7 +374,7 @@ plan_b (char const *filename)
     report_revision (found_revision);
   Fseek (ifp, 0, SEEK_SET);		/* rewind file */
   for (tibufsize = TIBUFSIZE_MINIMUM;  tibufsize < maxlen;  tibufsize <<= 1)
-    continue;
+    /* do nothing */ ;
   lines_per_buf = tibufsize / maxlen;
   tireclen = maxlen;
   tibuf[0] = xmalloc (2 * tibufsize);
@@ -454,7 +454,7 @@ ifetch (LINENUM line, bool whichbuf, size_t *psize)
 	    *psize = last_line_size;
 	else {
 	    for (q = p;  *q++ != '\n';  )
-		continue;
+		/* do nothing */ ;
 	    *psize = q - p;
 	}
 	return p;
diff --git a/src/maketime.c b/src/maketime.c
index 9084860..7c0f79b 100644
--- a/src/maketime.c
+++ b/src/maketime.c
@@ -389,7 +389,7 @@ maketime (struct partime const *pt, time_t default_time)
 	for (mon = 11;
 	     (tm.tm_mday = (yday2 - month_yday[mon] + (mon < 2 || !leap))) <= 0;
 	     mon--)
-	  continue;
+	  /* do nothing */ ;
 	tm.tm_mon = mon;
       }
 
diff --git a/src/merge.c b/src/merge.c
index bbd9d29..5948c46 100644
--- a/src/merge.c
+++ b/src/merge.c
@@ -425,7 +425,7 @@ merge_hunk (int hunk, struct outstate *outstate, LINENUM where,
 	   firstin < in && firstnew < new
 	     && context_matches_file (firstnew, lastwhere);
 	   firstin++, firstnew++, lastwhere++)
-	continue;
+	/* do nothing */ ;
       already_applied = (firstin == in && firstnew == new);
       if (already_applied)
 	merge_result (&first_result, hunk, "already applied",
@@ -458,7 +458,7 @@ merge_hunk (int hunk, struct outstate *outstate, LINENUM where,
 		   firstin < in && firstnew < new
 		   && context_matches_file (new - 1, lastwhere - 1);
 		   in--, new--, lastwhere--, common_suffix++)
-		continue;
+		/* do nothing */ ;
 	    }
 
 	  lines = 3 + (in - firstin) + (new - firstnew);
diff --git a/src/partime.c b/src/partime.c
index 99b0f2a..1528844 100644
--- a/src/partime.c
+++ b/src/partime.c
@@ -342,7 +342,7 @@ parse_prefix (char const *str, char const **patterns, struct partime *t)
       while ((s = parse_pattern_letter (s, c, t)) != 0);
 
       while (*pat++)
-	continue;
+	/* do nothing */ ;
     }
 
   return 0;
diff --git a/src/pch.c b/src/pch.c
index ab2183c..94aafe4 100644
--- a/src/pch.c
+++ b/src/pch.c
@@ -389,13 +389,13 @@ intuit_diff_type (bool need_header)
 	if (ISDIGIT (*s))
 	  {
 	    for (t = s + 1; ISDIGIT (*t) || *t == ',';  t++)
-	      continue;
+	      /* do nothing */ ;
 	    if (*t == 'd' || *t == 'c' || *t == 'a')
 	      {
 		for (t++;  ISDIGIT (*t) || *t == ',';  t++)
-		  continue;
+		  /* do nothing */ ;
 		for (; *t == ' ' || *t == '\t'; t++)
-		  continue;
+		  /* do nothing */ ;
 		if (*t == '\r')
 		  t++;
 		this_is_a_command = (*t == '\n');
@@ -435,14 +435,14 @@ intuit_diff_type (bool need_header)
 	else if (strnEQ(s, "Prereq:", 7))
 	  {
 	    for (t = s + 7;  ISSPACE ((unsigned char) *t);  t++)
-	      continue;
+	      /* do nothing */ ;
 	    revision = t;
 	    for (t = revision;  *t;  t++)
 	      if (ISSPACE ((unsigned char) *t))
 		{
 		  char const *u;
 		  for (u = t + 1;  ISSPACE ((unsigned char) *u);  u++)
-		    continue;
+		    /* do nothing */ ;
 		  if (*u)
 		    {
 		      char numbuf[LINENUM_LENGTH_BOUND + 1];
@@ -463,7 +463,7 @@ intuit_diff_type (bool need_header)
 	else
 	  {
 	    for (t = s;  t[0] == '-' && t[1] == ' ';  t += 2)
-	      continue;
+	      /* do nothing */ ;
 	    if (strnEQ(t, "--- ", 4))
 	      {
 		time_t timestamp = (time_t) -1;
@@ -1013,7 +1013,7 @@ another_hunk (enum diff difftype, bool rev)
 		    return -1;
 		}
 		for (s = buf;  *s && !ISDIGIT (*s);  s++)
-		  continue;
+		  /* do nothing */ ;
 		if (strnEQ(s,"0,0",3))
 		    remove_prefix (s, 2);
 		s = scan_linenum (s, &p_first);
@@ -1086,7 +1086,7 @@ another_hunk (enum diff difftype, bool rev)
 		  }
 		p_Char[p_end] = '=';
 		for (s = buf;  *s && ! ISDIGIT (*s);  s++)
-		  continue;
+		  /* do nothing */ ;
 		s = scan_linenum (s, &p_newfirst);
 		if (*s == ',')
 		  {
@@ -1752,7 +1752,7 @@ incomplete_line (void)
   if (getc (fp) == '\\')
     {
       while ((c = getc (fp)) != '\n'  &&  c != EOF)
-	continue;
+	/* do nothing */ ;
       return true;
     }
   else
@@ -2017,13 +2017,13 @@ get_ed_command_letter (char const *line)
   if (ISDIGIT (*p))
     {
       while (ISDIGIT (*++p))
-	continue;
+	/* do nothing */ ;
       if (*p == ',')
 	{
 	  if (! ISDIGIT (*++p))
 	    return 0;
 	  while (ISDIGIT (*++p))
-	    continue;
+	    /* do nothing */ ;
 	  pair = true;
 	}
     }
diff --git a/src/util.c b/src/util.c
index d04edbe..80ce54f 100644
--- a/src/util.c
+++ b/src/util.c
@@ -187,7 +187,7 @@ create_backup (char *to, struct stat *to_st, int *to_errno,
 	  for (o = t + tlen, olen = 0;
 	       o > t && ! ISSLASH (*(o - 1));
 	       o--)
-	    continue;
+	    /* do nothing */ ;
 	  olen = t + tlen - o;
 	  tlen -= olen;
 	  bakname = xmalloc (plen + tlen + blen + olen + slen + 1);
@@ -641,7 +641,7 @@ remove_prefix (char *p, size_t prefixlen)
 {
   char const *s = p + prefixlen;
   while ((*p++ = *s++))
-    continue;
+    /* do nothing */ ;
 }
 
 char *
@@ -1008,7 +1008,7 @@ replace_slashes (char *filename)
   char const *component_start;
 
   for (f = filename + FILE_SYSTEM_PREFIX_LEN (filename);  ISSLASH (*f);  f++)
-    continue;
+    /* do nothing */ ;
 
   component_start = f;
 
-- 
1.6.5

